### PR TITLE
DP-460 Upgrade to PHP 8.1 / Laravel 9

### PR DESCRIPTION
**Pull request by daniilly**

### CODE REVIEW

**Comment by anas-srikou** [comment]
@alexanderko @daniilly This set as "ready for review" in the spreadsheet but it is set to "Draft" here.
Could you let me know which is correct please?

**Comment by daniilly** [comment]
> @alexanderko @daniilly This set as "ready for review" in the spreadsheet but it is set to "Draft" here. Could you let me know which is correct please?

The one in the spreadsheet, sorry for the inconvenience

File: src/Components/MosquittoClient.php [hunk 1]
```diff
@@ -195,25 +197,52 @@ public function subscribe(array $payload)
...
+     * @param array $topics
+     * @param string $currentTopic
+     * @param string $message
+     *  
+     * @return \DreamFactory\Core\Utility\ServiceResponse
+     */
+    protected function sendServiceRequest($topics, $currentTopic, $message) {
+        // Retrieve service information
+        $service = array_by_key_value($topics, 'topic', $currentTopic, 'service');
+        Log::debug('[MQTT] Triggering service: ' . json_encode($service));
```
Comment (anas-srikou) [inline src/Components/MosquittoClient.php:231]: @daniilly Is this debug statement here necessary? if it was for testing purposes then please remove it

File: src/Components/MosquittoClient.php [hunk 2]
```diff
@@ -147,46 +162,33 @@ public function subscribe(array $payload)
...
-                        false);
-                $content = $rs->getContent();
+            $client = $this->client($this->getConfig());
+
+            $client->connect($this->connectionSettings, true);
+
+            Log::info('[MQTT] Connected to MQTT broker for subscription.');
+
+            $handler = function (MqttClient $client, string $topic, string $message) use ($topics) {
+                Log::debug('[MQTT] Received message on topic: ' . $topic . ' with payload ' . $message);
```
Comment (anas-srikou) [inline src/Components/MosquittoClient.php:172]: @daniilly Log::info would be a better choice here if not used for testing purpose

File: src/Components/MosquittoClient.php (hunk 1 above)
Comment (daniilly) [reply src/Components/MosquittoClient.php:231]: @anas-srikou This line of code was there before. We can change the logging level to `info` or remove it

Comment (anas-srikou) [reply src/Components/MosquittoClient.php:231]: @daniilly Please change it to info 🙏

**Review by anas-srikou — APPROVED** [review]

**Pull request merged**